oci/handler/directory.go

```diff
@@ -323,88 +426,154 @@ func (handle *DirectoryHandler) FetchImage(ctx context.Context, fullref, platfor
 		return err
 	}
 
-	config, err := img.RawConfigFile()
-	if err != nil {
-		return err
-	}
-
-	configName, err := img.ConfigName()
+	parsableManifest, err := idx.IndexManifest()
 	if err != nil {
 		return err
 	}
 
-	configPath := filepath.Join(
-		handle.path,
-		DirectoryHandlerConfigsDir,
-		configName.Algorithm,
-		configName.Hex,
-	)
-
-	// If the config already exists, skip it
-	if _, err := os.Stat(configPath); err == nil {
-		return nil
-	}
-
-	// Recursively create the directory
-	if err = os.MkdirAll(configPath[:strings.LastIndex(configPath, "/")], 0o775); err != nil {
-		return err
+	var manifests []v1.Descriptor
+	for _, descriptor := range parsableManifest.Manifests {
+		if descriptor.Platform.OS == OS && descriptor.Platform.Architecture == arch {
+			log.G(ctx).Tracef("Fetching %s/%s manifest %s",
+				descriptor.Platform.Architecture,
+				descriptor.Platform.OS,
+				descriptor.Digest)
+			manifests = append(manifests, descriptor)
+		} else {
+			log.G(ctx).Tracef("Skip fetching %s/%s manifest %s",
```

Comment:
Log messages should start with a lowercase letter. Let's also use `WithFields` too, for example:
```
log.G(ctx).
  WithField("platform", fmt.Sprintf("%s/%s", descriptor.Platform.OS, descriptor.Platform.Architecture)).
  WithField("digest", descriptor.Digest).
  Tracef("fetching")
```